pandas/tests/indexes/period/test_ops.py

```diff
@@ -2,24 +2,29 @@
 import pytest
 
 import pandas as pd
-from pandas import DatetimeIndex, Index, NaT, PeriodIndex, Series
+from pandas import DatetimeIndex, Index, NaT, PeriodIndex, Series, TimedeltaIndex
 from pandas.core.arrays import PeriodArray
-from pandas.tests.base.test_ops import Ops
+from pandas.tests.base.utils import check_ops_properties_valid
 import pandas.util.testing as tm
 
 
-class TestPeriodIndexOps(Ops):
-    def setup_method(self, method):
-        super().setup_method(method)
-        mask = lambda x: (isinstance(x, DatetimeIndex) or isinstance(x, PeriodIndex))
-        self.is_valid_objs = [o for o in self.objs if mask(o)]
-        self.not_valid_objs = [o for o in self.objs if not mask(o)]
+class TestPeriodIndexOps:
+    @pytest.mark.parametrize("op", PeriodArray._datetimelike_ops)
```

Comment:
i would rather actually either hardcode these ops and/or create a fixture. we already do a lot of this in pandas/tests/series/test_datetime_values.py where we test all accessors, so this is likely duplicative. (though its possible we are just not fully testing the Index ones).

I am not saying to actually fix this here, but to add appropriate TODOs / lists of things to come back around and patch.

Reply:
Would creating a follow-up ticket be enough?

Reply:
You're right, this is already tested in `pandas/tests/series/test_datetime_values.py::test_dt_namespace_accessor`. I'll remove my test cases then